src/mol.cpp

```diff
@@ -1176,29 +1176,26 @@ namespace OpenBabel
   }
 
   //! Returns the total spin multiplicity -- if it has not previously been set
-  //!  It is calculated from the atomic spin multiplicity information
-  //!  assuming the high-spin case (i.e. it simply sums the number of unpaired
-  //!  electrons assuming no further pairing of spins.
+  //! calculating from atomic formal charge
   unsigned int OBMol::GetTotalSpinMultiplicity()
   {
     if (HasFlag(OB_TSPIN_MOL))
       return(_totalSpin);
-    else // calculate from atomic spin information (assuming high-spin case)
+    else // calculate from total charge information
       {
         obErrorLog.ThrowError(__FUNCTION__,
-                              "Ran OpenBabel::GetTotalSpinMultiplicity -- calculating from atomic spins assuming high spin case",
+                              "Ran OpenBabel::GetTotalSpinMultiplicity -- calculating from atomic formal charge",
                               obAuditMsg);
-
         OBAtom *atom;
         vector<OBAtom*>::iterator i;
-        unsigned int unpaired_electrons = 0;
+        int chg = 0;
+        int spin = 0;
 
         for (atom = BeginAtom(i);atom;atom = NextAtom(i))
-          {
-            if (atom->GetSpinMultiplicity() > 1)
-              unpaired_electrons += (atom->GetSpinMultiplicity() - 1);
```

Comment:
Yeah, you can't just throw this out. There *are* molecules with atom spin set.

I'd be OK with something that checks if there are any atoms with spin multiplicity set and performing your calculation if nothing is a radical.